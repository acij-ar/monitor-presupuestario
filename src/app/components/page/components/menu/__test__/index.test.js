const React = require('react');
const { shallow } = require('enzyme');

const Menu = require('..');

describe('Menu component', () => {
  it('should match snapshot for active page budget', () => {
    const wrapper = shallow(<Menu selectedPage="budget" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapshot for active page monitor', () => {
    const wrapper = shallow(<Menu selectedPage="monitor" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapshot for active page about', () => {
    const wrapper = shallow(<Menu selectedPage="about" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapshot for active page doubts', () => {
    const wrapper = shallow(<Menu selectedPage="doubts" />);
    expect(wrapper).toMatchSnapshot();
  });
});
