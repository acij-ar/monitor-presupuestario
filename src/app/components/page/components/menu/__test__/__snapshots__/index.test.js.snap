// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu component should match snapshot for active page about 1`] = `
<div
  id="menu"
>
  <div
    id="menu-content"
  >
    <a
      href="/"
    >
      <img
        alt="Monitor Presupuestario"
        id="logo"
        src="/static/logo.svg"
        title="Monitor Presupuestario"
      />
    </a>
    <ul
      id="menu-sections"
    >
      <li>
        <a
          className=""
          href="/el-presupuesto"
          id="menu-link-budget"
        >
          El presupuesto
        </a>
        <SubMenu
          pageSections={
            Array [
              Object {
                "id": "what-is-the-budget",
                "name": "¿Qué es?",
              },
              Object {
                "id": "budget-cycle",
                "name": "El ciclo del presupuesto",
              },
              Object {
                "id": "budget-classifications",
                "name": "Clasificaciones",
              },
              Object {
                "id": "budget-spent",
                "name": "¿En qué se gasta?",
              },
              Object {
                "id": "budget-what-for",
                "name": "¿Para qué se gasta?",
              },
              Object {
                "id": "budget-who",
                "name": "¿Quién gasta?",
              },
              Object {
                "id": "budget-inflation",
                "name": "Ajuste de inflación",
              },
              Object {
                "id": "budget-analysis",
                "name": "Análisis presupuestario",
              },
            ]
          }
        />
      </li>
      <li>
        <a
          className=""
          href="/monitor"
          id="menu-link-monitor"
        >
          Monitor
        </a>
        <SubMenu
          pageSections={
            Array [
              Object {
                "href": "/monitor/explorar",
                "id": "explore-menu-link",
                "name": "Explorar",
              },
              Object {
                "href": "/monitor/comparar",
                "id": "compare-menu-link",
                "name": "Comparar",
              },
            ]
          }
        />
      </li>
      <li>
        <a
          className="active"
          href="/acerca-de"
          id="menu-link-about"
        >
          Acerca de
        </a>
        <SubMenu
          pageSections={
            Array [
              Object {
                "id": "about-main-content",
                "name": "El Monitor Presupuestario",
              },
              Object {
                "id": "about-acij",
                "name": "ACIJ",
              },
              Object {
                "id": "about-matchbox",
                "name": "Matchbox",
              },
              Object {
                "id": "about-methodology",
                "name": "Metodología",
              },
            ]
          }
        />
      </li>
      <li>
        <a
          className=""
          href="/dudas"
          id="menu-link-doubts"
        >
          Dudas
        </a>
      </li>
    </ul>
  </div>
</div>
`;

exports[`Menu component should match snapshot for active page budget 1`] = `
<div
  id="menu"
>
  <div
    id="menu-content"
  >
    <a
      href="/"
    >
      <img
        alt="Monitor Presupuestario"
        id="logo"
        src="/static/logo.svg"
        title="Monitor Presupuestario"
      />
    </a>
    <ul
      id="menu-sections"
    >
      <li>
        <a
          className="active"
          href="/el-presupuesto"
          id="menu-link-budget"
        >
          El presupuesto
        </a>
        <SubMenu
          pageSections={
            Array [
              Object {
                "id": "what-is-the-budget",
                "name": "¿Qué es?",
              },
              Object {
                "id": "budget-cycle",
                "name": "El ciclo del presupuesto",
              },
              Object {
                "id": "budget-classifications",
                "name": "Clasificaciones",
              },
              Object {
                "id": "budget-spent",
                "name": "¿En qué se gasta?",
              },
              Object {
                "id": "budget-what-for",
                "name": "¿Para qué se gasta?",
              },
              Object {
                "id": "budget-who",
                "name": "¿Quién gasta?",
              },
              Object {
                "id": "budget-inflation",
                "name": "Ajuste de inflación",
              },
              Object {
                "id": "budget-analysis",
                "name": "Análisis presupuestario",
              },
            ]
          }
        />
      </li>
      <li>
        <a
          className=""
          href="/monitor"
          id="menu-link-monitor"
        >
          Monitor
        </a>
        <SubMenu
          pageSections={
            Array [
              Object {
                "href": "/monitor/explorar",
                "id": "explore-menu-link",
                "name": "Explorar",
              },
              Object {
                "href": "/monitor/comparar",
                "id": "compare-menu-link",
                "name": "Comparar",
              },
            ]
          }
        />
      </li>
      <li>
        <a
          className=""
          href="/acerca-de"
          id="menu-link-about"
        >
          Acerca de
        </a>
        <SubMenu
          pageSections={
            Array [
              Object {
                "id": "about-main-content",
                "name": "El Monitor Presupuestario",
              },
              Object {
                "id": "about-acij",
                "name": "ACIJ",
              },
              Object {
                "id": "about-matchbox",
                "name": "Matchbox",
              },
              Object {
                "id": "about-methodology",
                "name": "Metodología",
              },
            ]
          }
        />
      </li>
      <li>
        <a
          className=""
          href="/dudas"
          id="menu-link-doubts"
        >
          Dudas
        </a>
      </li>
    </ul>
  </div>
</div>
`;

exports[`Menu component should match snapshot for active page doubts 1`] = `
<div
  id="menu"
>
  <div
    id="menu-content"
  >
    <a
      href="/"
    >
      <img
        alt="Monitor Presupuestario"
        id="logo"
        src="/static/logo.svg"
        title="Monitor Presupuestario"
      />
    </a>
    <ul
      id="menu-sections"
    >
      <li>
        <a
          className=""
          href="/el-presupuesto"
          id="menu-link-budget"
        >
          El presupuesto
        </a>
        <SubMenu
          pageSections={
            Array [
              Object {
                "id": "what-is-the-budget",
                "name": "¿Qué es?",
              },
              Object {
                "id": "budget-cycle",
                "name": "El ciclo del presupuesto",
              },
              Object {
                "id": "budget-classifications",
                "name": "Clasificaciones",
              },
              Object {
                "id": "budget-spent",
                "name": "¿En qué se gasta?",
              },
              Object {
                "id": "budget-what-for",
                "name": "¿Para qué se gasta?",
              },
              Object {
                "id": "budget-who",
                "name": "¿Quién gasta?",
              },
              Object {
                "id": "budget-inflation",
                "name": "Ajuste de inflación",
              },
              Object {
                "id": "budget-analysis",
                "name": "Análisis presupuestario",
              },
            ]
          }
        />
      </li>
      <li>
        <a
          className=""
          href="/monitor"
          id="menu-link-monitor"
        >
          Monitor
        </a>
        <SubMenu
          pageSections={
            Array [
              Object {
                "href": "/monitor/explorar",
                "id": "explore-menu-link",
                "name": "Explorar",
              },
              Object {
                "href": "/monitor/comparar",
                "id": "compare-menu-link",
                "name": "Comparar",
              },
            ]
          }
        />
      </li>
      <li>
        <a
          className=""
          href="/acerca-de"
          id="menu-link-about"
        >
          Acerca de
        </a>
        <SubMenu
          pageSections={
            Array [
              Object {
                "id": "about-main-content",
                "name": "El Monitor Presupuestario",
              },
              Object {
                "id": "about-acij",
                "name": "ACIJ",
              },
              Object {
                "id": "about-matchbox",
                "name": "Matchbox",
              },
              Object {
                "id": "about-methodology",
                "name": "Metodología",
              },
            ]
          }
        />
      </li>
      <li>
        <a
          className="active"
          href="/dudas"
          id="menu-link-doubts"
        >
          Dudas
        </a>
      </li>
    </ul>
  </div>
</div>
`;

exports[`Menu component should match snapshot for active page monitor 1`] = `
<div
  id="menu"
>
  <div
    id="menu-content"
  >
    <a
      href="/"
    >
      <img
        alt="Monitor Presupuestario"
        id="logo"
        src="/static/logo.svg"
        title="Monitor Presupuestario"
      />
    </a>
    <ul
      id="menu-sections"
    >
      <li>
        <a
          className=""
          href="/el-presupuesto"
          id="menu-link-budget"
        >
          El presupuesto
        </a>
        <SubMenu
          pageSections={
            Array [
              Object {
                "id": "what-is-the-budget",
                "name": "¿Qué es?",
              },
              Object {
                "id": "budget-cycle",
                "name": "El ciclo del presupuesto",
              },
              Object {
                "id": "budget-classifications",
                "name": "Clasificaciones",
              },
              Object {
                "id": "budget-spent",
                "name": "¿En qué se gasta?",
              },
              Object {
                "id": "budget-what-for",
                "name": "¿Para qué se gasta?",
              },
              Object {
                "id": "budget-who",
                "name": "¿Quién gasta?",
              },
              Object {
                "id": "budget-inflation",
                "name": "Ajuste de inflación",
              },
              Object {
                "id": "budget-analysis",
                "name": "Análisis presupuestario",
              },
            ]
          }
        />
      </li>
      <li>
        <a
          className=""
          href="/monitor"
          id="menu-link-monitor"
        >
          Monitor
        </a>
        <SubMenu
          pageSections={
            Array [
              Object {
                "href": "/monitor/explorar",
                "id": "explore-menu-link",
                "name": "Explorar",
              },
              Object {
                "href": "/monitor/comparar",
                "id": "compare-menu-link",
                "name": "Comparar",
              },
            ]
          }
        />
      </li>
      <li>
        <a
          className=""
          href="/acerca-de"
          id="menu-link-about"
        >
          Acerca de
        </a>
        <SubMenu
          pageSections={
            Array [
              Object {
                "id": "about-main-content",
                "name": "El Monitor Presupuestario",
              },
              Object {
                "id": "about-acij",
                "name": "ACIJ",
              },
              Object {
                "id": "about-matchbox",
                "name": "Matchbox",
              },
              Object {
                "id": "about-methodology",
                "name": "Metodología",
              },
            ]
          }
        />
      </li>
      <li>
        <a
          className=""
          href="/dudas"
          id="menu-link-doubts"
        >
          Dudas
        </a>
      </li>
    </ul>
  </div>
</div>
`;
